### DISCUSSION TITLE
Use picture tag more often, to support more formats

### DISCUSSION THREAD

**Discussion by BrocksiNet** [shopware/platform#3151] [RFC / Proposal]
**Effort**: middle

### Description (story telling): 
You as a shop-manager wanna create a nice homepage. This means you will add a big picture about some _super-duppa_ product on your page. But after that your SEO's will tell you that your web vitals are super bad (user downloads over 1MB big pictures). Then you will research and think about using a better format like webp or avif. But then you will realize that you can only use one image in the administration for a single (different) content elements. This makes you really sad, because you would like to use [avif](https://caniuse.com/?search=avif) but [webp](https://caniuse.com/?search=webp) is more supported.

**Simple example code:**
```
<picture>  
    <source type="image/avif" srcset="bunny.avif">  
    <img alt="Bunnies are cute!" src="bunny.jpg">
</picture>
```

### Benefits
We should not think that a image is only one element. An image can be multiple elements with the use of the picture tag. So shopware should prepare for that and find a dynamic way of adding multiple different files (formats) and attributes (media="(min-width:465px)"). Nobody know what kind of format we will use in the future so this is useful and also in combination with flysystem or external image services it can be a great improve to the Shopping Experiences (providing multiple images in different formats/sizes).

### Break strategy
Simple, only use picture tag when more then one images are provided in cms, product, category and so on.

**Comment by tobiasberge** [comment]
Hi @bjoern-flagbit,

I am not sure if I understand this correctly, maybe you can go a bit more into detail.

Do you suggest the usage of the picture element to serve the "same" image, but in different formats? Like inside the code snippet? And the core would theoretically convert the images into formats like avif or webp automatically (even when e.g. a .jpeg was uploaded?) Or should the user somehow control this manually to have more flexibility?

Or do you suggest to use the media query aspect of picture to serve a different image format? (e.g. when viewport above x pixel use different image format) But I'm not sure if this would be beneficial. I think this would interfere with the current solution where srcset is used to load different thumbnail sizes depending on the viewport. In theory this could be re-written to use picture instead. But imo srcset is enough to serve the thumbnail sizes of the same image.

From my understanding it would look like this then. Thumbnails still handled via srcset and picture wrapper for potentially other img formats:

```html
<picture>
  <source type="image/avif"
       srcset="
       thumbnail_1920x1920.avif 1920w, 
       thumbnail_800x800.avif 800w, 
       thumbnail_400x400.avif 400w
       " 
       sizes="310px" 
  />
  <img src="image.jpg"
       srcset="
       thumbnail_1920x1920.jpg 1920w, 
       thumbnail_800x800.jpg 800w, 
       thumbnail_400x400.jpg 400w
       " 
       sizes="310px" 
  />
</picture>
```

**Comment by tinect** [comment]
I really don't like the idea ;-)

Instead of cluttering up your hard drives, use a service that serves every supported format dynamically.
This saves: time, energy, co2, burn-out and other funny stuff.

I'd rather suggest shopware hosts something like imgproxy for shops and implement this feature into core: https://github.com/FriendsOfShopware/FroshPlatformThumbnailProcessor
